scripts/plot_scalar_resources.py

```diff
@@ -0,0 +1,137 @@
+r"""
+Inputs
+-------
+resources : str
+    path of input file with data in scalar resources template format as .csv
+target : str
+    path of output file with prepared data as .csv
+
+Outputs
+---------
+matplotlib.figure.Figure
+    plotted scalar resources
+
+Description
+-------------
+The script takes data in the format of the scalar resources template, prepares the
+data and plots the data.
+"""
+import sys
+import os
+import pandas as pd
+import matplotlib.pyplot as plt
+import oemoflex.tools.plots as plots
+from oemoflex.tools.plots import colors_odict
+
+if __name__ == "__main__":
+    resources = sys.argv[1]
+
+    target = sys.argv[2]
+
+    target_dir = os.path.dirname(target)
+
+    # create the directory plotted where all plots are saved
+    if not os.path.exists(target_dir):
+        os.makedirs(target_dir)
+
+    def load_scalars(path):
+        df = pd.read_csv(path, sep=",", index_col=0)
+        return df
+
+    # Load scalar data
+    scalars = load_scalars(resources)
+
+    # GENERAL
+    unit_dict = {"capacity_net_el": "W"}
+    carrier_dict = {
+        "biomass": "Biomass",
+        "ch4": "CH4",
+        "hard coal": "Hard coal",
+        "oil": "Oil",
+        "lignite": "Lignite",
+        "other": "Other",
+    }
+
+    german_labels = {
+        "Biomass": "Biomasse",
+        "CH4": "Erdgas",
+        "Hard coal": "Steinkohle",
+        "Oil": "Öl",
+        "Lignite": "Braunkohle",
+        "Other": "Andere",
+    }
+
+    # User input
+    var_name = "capacity_net_el"
+    conv_number = 1000 * 1000
+
+    def prepare_conv_pp_scalars(var_name, conv_number, carrier_dict=carrier_dict):
+        # select var_name to be plotted
+        selected_df = scalars[scalars["var_name"] == var_name].copy()
+
+        # CONV_PP specific
+        selected_df.loc[(selected_df["region"] != "Berlin"), "region"] = "Brandenburg"
+        # capitalize carrier names
+        selected_df["carrier"].replace(carrier_dict, inplace=True)
+        # aggregate carriers in regions
+        selected_df_agg = selected_df.groupby(
+            ["scenario", "region", "carrier", "var_name"]
+        ).sum()
+        selected_df_agg.reset_index(inplace=True)
+
+        # apply pivot table
+        df_pivot = pd.pivot_table(
+            selected_df_agg,
+            index=["region"],
+            columns="carrier",
+            values="var_value",
+        )
+
+        # colors
+        color_keys = df_pivot.columns
```

Comment:
Would be good to make clear where the translation from English to German happens, so that it can be switched on and off easily.